pilot/pkg/bootstrap/configcontroller.go

```diff
@@ -471,3 +489,41 @@ func (s *Server) getRootCertFromSecret(name, namespace string) (*istioCredential
 	}
 	return kube.ExtractRoot(secret.Data)
 }
+
+// checkAndRunNonRevisionLeaderElectionIfRequired checks the ConfigMap for leader election and runs non-revision leader election
+// to figure out when the older leader has released the lock.
+// Once the lock is acquired, it will close the activateCh channel to signal the per-revision leader to start writing status.
+func (s *Server) checkAndRunNonRevisionLeaderElectionIfRequired(args *PilotArgs, activateCh chan struct{}) {
+	cm, err := s.kubeClient.Kube().CoreV1().ConfigMaps(args.Namespace).Get(context.Background(), leaderelection.GatewayStatusController, v1.GetOptions{})
+	if err != nil {
+		// ConfigMap does not exist, so per-revision leader election should be active
+		close(activateCh)
+		return
+	}
+	leaderAnn, ok := cm.Annotations[k8sresourcelock.LeaderElectionRecordAnnotationKey]
+	if ok {
+		var leaderInfo struct {
+			HolderIdentity string `json:"holderIdentity"`
+		}
+		if err := json.Unmarshal([]byte(leaderAnn), &leaderInfo); err == nil {
+			if leaderInfo.HolderIdentity != "" {
+				// Non-revision leader election should run, per-revision should be waiting for activation
+				s.addTerminatingStartFunc("gateway status", func(stop <-chan struct{}) error {
+					leaderelection.
+						NewLeaderElection(args.Namespace, args.PodName, leaderelection.GatewayStatusController, args.Revision, s.kubeClient).
```

Comment:
nit: instead of the same code copied twice it would be nice to just have the NewLeaderElection vs NewPerRevisionLeaderElection part changed. Else they are very likely to fall out of sync